secure_software_development_fundamentals.md

```diff
@@ -4499,7 +4499,18 @@ Many symmetric key algorithms, including AES, are what is called *block algorith
 
 The ECB mode is basically a debug or test mode for testing cryptographic algorithms. In ECB mode, the same block of data will produce the same encryption result. This is disastrous for an encryption algorithm, because it reveals far too much about the data that is supposed to be encrypted. A great illustration of this is the so-called “ECB Penguin” image; this image is encrypted using an ECB mode. Encrypted images should appear as random noise, but because ECB mode is used, in the ECB Penguin the image of Tux the Penguin is clearly visible.
```

Comment:
Presuming we do this, the paragraph will need to explain the GCM penguin too. E.g., replace the last sentence with something like:

> Note that in the ECB Penguin below, the image of Tux the Penguin is clearly (and disastrously) visible. An encrypted image *should* appear as random noise, as illustrated by the GCM Penguin below (an encrypted image that uses Galois/Counter mode (GCM)).

Reply:
Updated. I've avoided the double parenthesis in the latter sentence if that's ok.